stream-chat-android-client/src/main/java/io/getstream/chat/android/client/di/BaseChatModule.kt

```diff
@@ -134,21 +150,23 @@ internal open class BaseChatModule(
     private fun buildApi(chatConfig: ChatClientConfig): ChatApi {
         return ChatApi(
             chatConfig.apiKey,
-            buildRetrofitApi(),
+            buildRetrofitApi(RetrofitApi::class.java, false),
+            buildRetrofitApi(RetrofitAnonymousApi::class.java, true),
```

Comment:
We're passing in the same information - whether this is the anonymous API or not - twice here. Would be better to avoid this duplication.

Reply:
No, they are two different interfaces, one with the authenticated API endpoints and another with the anonymous API endpoints

Reply:
I know, but we're passing in the same logical information - whether this is the authenticated API or the anonymous API being built - twice. Both parameters are telling the method the same thing.

And this allows for invalid sets of parameter values, for example calling the method like this makes no sense, but is technically possible now:

```kotlin
buildRetrofitApi(RetrofitAnonymousApi::class.java, false)
```

Having just a single parameter would avoid this.

Reply:
Conversation resulted in https://github.com/GetStream/stream-chat-android/pull/767